### PR TITLE
BP-1034 - Subscriber to redis returns nothing

### PR DESCRIPTION
**Pull request by erezz-mov-ai**
[BP-1034](https://movai.atlassian.net/browse/BP-1034) - Subscriber to redis returns nothing

[BP-1034]: https://movai.atlassian.net/browse/BP-1034?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: dal/classes/protocols/wsredissub.py [hunk 1]
```diff
@@ -6,6 +6,7 @@
    Developers:
    - Manuel Silva (manuel.silva@mov.ai) - 2020
    - Tiago Paulino (tiago@mov.ai) - 2020
+   - Dor Marcus (dor@mov.ai) - 2023
```
Comment (engdorm) [inline dal/classes/protocols/wsredissub.py:9]: thanks for the recognition!

File: dal/classes/protocols/wsredissub.py [hunk 2]
```diff
@@ -164,6 +141,29 @@ async def read_websocket_loop(
...
+            ws_resp: websocket _response
+            connection_queue: queue to write messages to Websocket
+        """
+        while True:
+            msg = await connection_queue.get()
+            try:
+                if ws_resp is not None and not ws_resp.closed and not ws_resp._closing:
+                    await ws_resp.send_json(msg)
+                else:
+                    break
```
[the remark this answers is not in the document]
Comment (engdorm) [reply dal/classes/protocols/wsredissub.py:163]: before the break of course 
or even `del connection_queue`
not to create memory leak

File: dal/classes/protocols/wsredissub.py [hunk 3]
```diff
@@ -195,7 +195,8 @@ async def add_pattern(self, conn_id, conn, _pattern, **ignore):
         for key_pattern in key_patterns:
             pattern = "__keyspace@*__:%s" % (key_pattern)
             channel = await conn.psubscribe(pattern)
-            asyncio.create_task(self.wait_message(conn_id, channel[0]))
+            task = asyncio.create_task(self.wait_message(conn_id, channel[0]))
+            self.tasks[conn_id].append(task)
```
Comment (engdorm) [inline dal/classes/protocols/wsredissub.py:199]: please init this var in __init__ (self.tasks)

**Review by engdorm — CHANGES_REQUESTED** [review]
and run black

File: dal/classes/protocols/wsredissub.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (erezz-mov-ai) [reply dal/classes/protocols/wsredissub.py:164]: done

File: dal/classes/protocols/wsredissub.py (hunk 3 above)
Comment (erezz-mov-ai) [reply dal/classes/protocols/wsredissub.py]: I have removed the tasks list, it's not necessary

**Review by engdorm — APPROVED** [review]

**Pull request merged**